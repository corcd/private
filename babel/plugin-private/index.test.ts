/*
 * @Author: Whzcorcd
 * @Date: 2021-11-07 16:57:53
 * @LastEditors: Whzcorcd
 * @LastEditTime: 2021-11-07 21:51:42
 * @Description: file content
 */
const babel = require('@babel/core')
const p = require('./index')

const testCode = `
  wrapPrivate.include()
  wrapPrivate.exclude()

  wrapPrivate.include(["prod", "cmcc", "vvku"])
  wrapPrivate.exclude(["prod", "cmcc", "vvku"])

  function foo(){
    wrapPrivate.exclude(["prod", "cmcc", "vvku"], () => {
      function c() {}
      c()
    })
  }
  const score = ()=> {
    wrapPrivate.include(["prod", "cmcc", "vvku"], () => {})
  }
  const bar = ()=> {
    wrapPrivate.include(["prod", "cmcc", "vvku"], () => {
      const temp = 15214
      console.log(temp)
      const a = temp++
    })
  }
  const fooBar = wrapPrivate
  fooBar()
`;

it('Test privatePlugin', () => {
  const { code } = babel.transform(testCode, {
    plugins: [p],
  })
  expect(code).toMatchSnapshot()
})