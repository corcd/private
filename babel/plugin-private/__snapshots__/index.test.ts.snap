// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test privatePlugin 1`] = `
"function foo() {}

const score = () => {};

const bar = () => {
  const temp = 15214;
  console.log(temp);
  const a = temp++;
};

const fooBar = wrapPrivate;
fooBar();"
`;
